### PR TITLE
Feature: Search improvements

### PR DESCRIPTION
**Pull request by miagilepner**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: stream_chat/tests/test_client.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 from stream_chat.base.exceptions import StreamAPIException
 
 
-@pytest.mark.incremental
```
Comment (ferhatelmas) [inline stream_chat/tests/test_client.py:12]: since passes, looks like a good idea

File: stream_chat/client.py [hunk 2]
```diff
@@ -294,8 +294,18 @@ def get_rate_limits(
...
+            if sort or "next" in options:
+                raise ValueError("cannot use offset with sort or next parameters")
+        params = options.copy()
+        if isinstance(query, str):
+            params.update({"query": query})
+        else:
+            params.update({"message_filter_conditions": query})
+        params.update(
+            {"filter_conditions": filter_conditions, "sort": self.normalize_sort(sort)}
+        )
```
Comment (ferhatelmas) [inline stream_chat/client.py:308]: would be nice to have a helper to prepare params then duplicate

File: stream_chat/tests/test_client.py [hunk 3]
```diff
@@ -248,6 +247,33 @@ def test_get_rate_limits(self, client):
             > response["server_side"]["GetRateLimits"]["remaining"]
         )
 
+    @pytest.mark.xfail
+    def test_search_with_sort(self, client, channel, random_user):
```
Comment (ferhatelmas) [inline stream_chat/tests/test_client.py:251]: could you provide same test for async client?

File: stream_chat/tests/test_client.py (hunk 1 above)
Comment (miagilepner) [reply stream_chat/tests/test_client.py:13]: you can't use `@pytest.mark.xfail` with this

Comment (ferhatelmas) [reply stream_chat/tests/test_client.py:13]: could you put a link here for docs?

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**